### PR TITLE
fixed rapid fire

### PR DESCRIPTION
**Pull request by Sys0xdbg**

### CODE REVIEW

**Comment by ghost** [comment]
Like what’d you do tho

**Comment by Sys0xdbg** [comment]
> Like what’d you do tho

What?

**Comment by xiaoxiao921** [comment]
Explain what this PR do and what is the behaviour change

**Comment by Sys0xdbg** [comment]
> Explain what this PR do and what is the behaviour change

Well it's rapid fire. Real rapid fire. Not fake one that uses cam.

**Comment by ghost** [comment]
> > Explain what this PR do and what is the behaviour change
> 
> 
> 
> Well it's rapid fire. Real rapid fire. Not fake one that uses cam. 

Thats epic if it was implemented how I think it was.

**Comment by Sys0xdbg** [comment]
> > > Explain what this PR do and what is the behaviour change
> > 
> > 
> > Well it's rapid fire. Real rapid fire. Not fake one that uses cam.
> 
> Thats epic if it was implemented how I think it was.

https://gifdb.com/images/high/thinking-emoji-face-fwtb8bb2mnav8xjq.gif

**Comment by lonelybud** [comment]
>Well it's rapid fire. Real rapid fire. Not fake one that uses cam.

thats good but the previous dev must have done that intentionally? have you checked the history.

**Comment by xiaoxiao921** [comment]
> > Well it's rapid fire. Real rapid fire. Not fake one that uses cam.
> 
> thats good but the previous dev must have done that intentionally? have you checked the history.

I don't think there was any particular reason, just that someone went with a particular impl and everyone tried to improv it

**Comment by Sys0xdbg** [comment]
> > Well it's rapid fire. Real rapid fire. Not fake one that uses cam.
> 
> thats good but the previous dev must have done that intentionally? have you checked the history.

lol bro don't make it complicate. its just rapid fire.

**Comment by lonelybud** [comment]
does not work on molotov

**Comment by xiaoxiao921** [comment]
if else branch could be made for those weapons, or we just don't merge it

**Comment by ghost** [comment]
I think the current rapid fire shooting from the camera is fine, it just needs to be kind of tweaked so when shooting with a controller in the car it doesn’t shoot while you’re driving as sometimes it get confused. Idk it’s hard to explain, especially while im tired. But ya, I assume rapid fire is intended to shoot from the camera so that projectiles work, that’s probably the only way but what do I know.

File: src/backend/looped/weapons/rapid_fire.cpp [hunk 1]
```diff
@@ -14,50 +14,61 @@ namespace big
...
-					        -1,
-					        0,
-					        7);
-					int did_raycast_hit = 0;
-					Vector3 raycast_hit_position{};
-					Vector3 raycast_surface_normal_hit_position{};
-					Entity raycast_hit_entity{};
-					SHAPETEST::GET_SHAPE_TEST_RESULT(raycast_handle, &did_raycast_hit, &raycast_hit_position, &raycast_surface_normal_hit_position, &raycast_hit_entity);
-					if (did_raycast_hit)
+				if (WEAPON::GET_WEAPONTYPE_GROUP(weapon_hash) == 1548507267)
```
Comment (gir489returns) [inline src/backend/looped/weapons/rapid_fire.cpp:29]: Replace this with GROUP_THROWN from weapons.hpp.

File: src/backend/looped/weapons/rapid_fire.cpp [hunk 2]
```diff
@@ -14,50 +14,61 @@ namespace big
...
-					MISC::SHOOT_SINGLE_BULLET_BETWEEN_COORDS(weapon_position.x, weapon_position.y, weapon_position.z, end.x, end.y, end.z, WEAPON::GET_WEAPON_DAMAGE(weapon_hash, 0), true, weapon_hash, self::ped, true, false, -1.0);
+						MISC::SHOOT_SINGLE_BULLET_BETWEEN_COORDS(weapon_position.x, weapon_position.y, weapon_position.z, end.x, end.y, end.z, WEAPON::GET_WEAPON_DAMAGE(weapon_hash, 0), true, weapon_hash, self::ped, true, false, -1.0);
+					}
 				}
+				else
+				{
+					if (PAD::IS_DISABLED_CONTROL_PRESSED(0, (int)ControllerInputs::INPUT_ATTACK) && PED::GET_PED_CONFIG_FLAG(self::ped, 78, 0))
+					{
+						PED::SET_PED_SHOOTS_AT_COORD(self::ped, end.x, end.y, end.z, true);
+						WEAPON::REFILL_AMMO_INSTANTLY(self::ped);
```
Comment (gir489returns) [inline src/backend/looped/weapons/rapid_fire.cpp:68]: Did the user ask for this?

**Review by gir489returns — CHANGES_REQUESTED** [review]

File: src/backend/looped/weapons/rapid_fire.cpp (hunk 2 above)
Comment (Sys0xdbg) [reply src/backend/looped/weapons/rapid_fire.cpp:68]: what user?

Comment (gir489returns) [reply src/backend/looped/weapons/rapid_fire.cpp:68]: The user of the cheat. You are basically reduplicating infinite clip which the cheat already has.

**Comment by Sys0xdbg** [comment]
soo much buggy

**Pull request closed without merging**